src/node_crypto.cc

```diff
@@ -3583,7 +3583,7 @@ SignBase::Error Sign::SignFinal(const char* key_pem,
   if (pkey == nullptr)
     goto exit;
 
-  if (EVP_SignFinal(&mdctx_, *sig, sig_len, pkey))
+  if (EVP_SignFinal(&mdctx_, *sig, sig_len, pkey) && 0 == ERR_peek_error())
```

Comment:
@indutny Moving `ERR_peek_error()` and `pkey == nullptr` checks to `PEM_read_bio_PrivateKey ` would dictate modifications to `deps/openssl` which is just an external dependency that we copied from other projects like `deps/v8`. I am not sure if a change to a dependency is desirable.

For the explanation for this test: The correct header for this `test_bad_rsa_privkey.pem` should have been `-----BEGIN PRIVATE KEY-----` and `-----END PRIVATE KEY-----` instead of `-----BEGIN RSA PRIVATE KEY-----` and `-----END RSA PRIVATE KEY-----`. I have attached some detailed information below.


> BEGIN RSA PRIVATE KEY is PKCS#1 and is just an RSA key. It is essentially just the key object from PKCS#8, but without the version or algorithm identifier in front. BEGIN PRIVATE KEY is PKCS#8 and indicates that the key type is included in the key data itself. From the link:
> 
> The unencrypted PKCS#8 encoded data starts and ends with the tags:
> 
> -----BEGIN PRIVATE KEY-----
> BASE64 ENCODED DATA
> -----END PRIVATE KEY-----
> Within the base64 encoded data the following DER structure is present:
> 
> PrivateKeyInfo ::= SEQUENCE {
>   version         Version,
>   algorithm       AlgorithmIdentifier,
>   PrivateKey      BIT STRING
> }
> 
> AlgorithmIdentifier ::= SEQUENCE {
>   algorithm       OBJECT IDENTIFIER,
>   parameters      ANY DEFINED BY algorithm OPTIONAL
> }
> So for an RSA private key, the OID is 1.2.840.113549.1.1.1 and there is a RSAPrivateKey as the PrivateKey key data bitstring.
> As opposed to BEGIN RSA PRIVATE KEY, which always specifies an RSA key and therefore doesn't include a key type OID. BEGIN RSA PRIVATE KEY is PKCS#1:
> 
> RSA Private Key file (PKCS#1)
> 
> The RSA private key PEM file is specific for RSA keys.
> 
> It starts and ends with the tags:
> 
> -----BEGIN RSA PRIVATE KEY-----
> BASE64 ENCODED DATA
> -----END RSA PRIVATE KEY-----